week_2/pet.rb

```diff
@@ -110,6 +118,15 @@ def self.cost_to_feed(pets, days)
   #
   # Note - Order is not important
   def self.group_by_animal_type(pets)
-    raise NotImplementedError # TODO
+    ans = {}
+    for pet in pets
```

Comment:
`each` is preferred way of [iterating](https://dev.to/jolouie7/ruby-each-map-reduce-unless-and-until-3b5c) in Ruby.